test/unit/core/write_concern_error.test.js

```diff
@@ -159,4 +161,51 @@ describe('WriteConcernError', function () {
       });
     });
   });
+
+  describe('errInfo property', () => {
+    let client;
+
+    beforeEach(async function () {
+      client = this.configuration.newClient({ monitorCommands: true });
+      await client.connect();
+    });
+
+    afterEach(async () => {
+      if (client) {
+        await client.close();
+        client.removeAllListeners();
+      }
+    });
+
+    it('should always be accessible', {
+      metadata: { requires: { mongodb: '>=5.0.0' } },
+      async test() {
+        try {
+          await client.db().collection('wc_details').drop();
+        } catch {
+          // don't care
+        }
+
+        const collection = await client
+          .db()
+          .createCollection('wc_details', { validator: { x: { $type: 'string' } } });
+
+        const evCapture = once(client, 'commandSucceeded');
+
+        try {
+          await collection.insertOne({ x: /not a string/ });
+          expect.fail('The insert should fail the validation that x must be a string');
+        } catch (error) {
+          expect(error).to.be.instanceOf(MongoServerError);
+          expect(error).to.have.property('code', 121);
+          expect(error).to.have.property('errInfo').that.is.an('object');
+        }
+
+        const commandSucceededEvents = await evCapture;
+        expect(commandSucceededEvents).to.have.lengthOf(1);
+        const ev = commandSucceededEvents[0];
+        expect(ev).to.have.nested.property('reply.writeErrors[0].errInfo').that.is.an('object');
```

Comment:
do we want to check to make sure that this errInfo is deep equal to the caught error's errInfo above?

Reply:
Oo yes that would be good